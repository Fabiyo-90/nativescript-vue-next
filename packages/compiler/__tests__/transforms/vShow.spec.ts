// todo: uncomment when vue-next#907 is released
test.todo('uncomment when vue-next#907 is released')
// import {
//   baseParse as parse,
//   transform,
//   generate,
//   CompilerOptions,
//   transformElement
// } from '@vue/compiler-core'
// import { transformShow } from '../../src/transforms/vShow'
// import { DOMErrorCodes } from '../../src/errors'
//
// function transformWithShow(template: string, options: CompilerOptions = {}) {
//   const ast = parse(template)
//   transform(ast, {
//     nodeTransforms: [transformElement],
//     directiveTransforms: {
//       show: transformShow
//     },
//     ...options
//   })
//   return ast
// }
//
// describe('compiler: v-show transform', () => {
//   test('simple expression', () => {
//     const ast = transformWithShow(`<Label v-show="a"/>`)
//
//     expect(generate(ast).code).toMatchSnapshot()
//   })
//
//   test('should raise error if has no expression', () => {
//     const onError = jest.fn()
//     transformWithShow(`<Label v-show/>`, { onError })
//
//     expect(onError).toHaveBeenCalledTimes(1)
//     expect(onError).toHaveBeenCalledWith(
//       expect.objectContaining({
//         code: DOMErrorCodes.X_V_SHOW_NO_EXPRESSION
//       })
//     )
//   })
// })
