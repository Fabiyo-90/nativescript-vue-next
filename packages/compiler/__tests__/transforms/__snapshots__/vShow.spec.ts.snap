// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler: v-show transform simple expression 1`] = `
"const _Vue = Vue

return function render(_ctx, _cache) {
  with (_ctx) {
    const { resolveComponent: _resolveComponent, vShow: _vShow, withDirectives: _withDirectives, openBlock: _openBlock, createBlock: _createBlock } = _Vue

    const _component_Label = _resolveComponent(\\"Label\\")

    return _withDirectives((_openBlock(), _createBlock(_component_Label, null, null, 512 /* NEED_PATCH */)), [
      [_vShow, a]
    ])
  }
}"
`;
